### PR TITLE
feat(extensions): Implement initial simple extension yaml reading

### PR DESCRIPTION
**Pull request by zeroshade**

### CODE REVIEW

**Comment by zeroshade** [comment]
@cpcloud @JamesRTaylor any luck getting a review / approval here?

**Comment by zeroshade** [comment]
rebased

**Comment by JamesRTaylor** [comment]
What are your thoughts on using `github.com/goccy/go-yaml` vs `gopkg.in/yaml.v3`?

File: extensions/simple_extension_test.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/substrait-io/substrait-go/extensions"
+
+	//"gopkg.in/yaml.v3"
```
Comment (JamesRTaylor) [inline extensions/simple_extension_test.go:13]: Remove commented code

File: go.sum [hunk 2]
```diff
@@ -13,6 +24,12 @@ github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
 github.com/kr/text v0.2.0 h1:5Nx0Ya0ZqY2ygV366QzturHI13Jq95ApcVaJBhpS+AY=
 github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
+github.com/leodido/go-urn v1.2.0 h1:hpXL4XnriNwQ/ABnpepYM/1vCLWNDfUNts8dX3xTG6Y=
+github.com/leodido/go-urn v1.2.0/go.mod h1:+8+nEpDfqqsY+g338gtMEUOtuK+4dEMhiQEgxpxOKII=
+github.com/mattn/go-colorable v0.1.8 h1:c1ghPdyEDarC70ftn0y+A/Ee++9zz8ljHG1b13eJ0s8=
+github.com/mattn/go-colorable v0.1.8/go.mod h1:u6P/XSegPjTcexA+o6vUJrdnUu04hMope9wVRipJSqc=
+github.com/mattn/go-isatty v0.0.12 h1:wuysRhFDzyxgEmMf5xjvJ2M9dZoWAXNNr5LSBS7uHXY=
+github.com/mattn/go-isatty v0.0.12/go.mod h1:cbi8OIDigv2wuxKPP5vlRcQ1OAZbq2CE4Kysco4FUpU=
```
Comment (JamesRTaylor) [inline go.sum:32]: Are all of these `go.sum` changes required?

Comment (zeroshade) [reply go.sum:32]: They are what I get when I run `go mod tidy` most likely they are the transitive dependencies for something in the `go-yaml` package.

**Comment by zeroshade** [comment]
> What are your thoughts on using `github.com/goccy/go-yaml` vs `gopkg.in/yaml.v3`?

Honestly, I prefer the API for `gopkg.in/yaml.v3`, but there are edge cases that it doesn't handle such as multi-document files or more specifically handling the 

```yaml
%YAML 1.2
---
```

prefix that is on top of the sample yamls in the substrait repository. So I went with go-yaml since it ends up being more performant but ultimately because it handles the form of the files in the examples without special handling.

**Pull request merged**